src/services/doppler-billing-api-client.ts

```diff
@@ -0,0 +1,106 @@
+import { ResultWithoutExpectedErrors } from '../doppler-types';
+import { AxiosInstance, AxiosStatic } from 'axios';
+import { AppSession } from './app-session';
+import { RefObject } from 'react';
+
+export interface DopplerBillingApiClient {
+  getInvoices(page: number, pageSize: number): Promise<ResultWithoutExpectedErrors<Invoices>>;
+}
+
+interface DopplerBillingApiConnectionData {
+  jwtToken: string;
+  idUser: number;
+}
+
+export interface Invoices {
+  items: Invoice[];
+  totalItems: number;
+}
+
+export interface Invoice {
+  accountId: string;
+  product: string;
+  date: Date;
+  currency: string;
+  amount: number;
+  filename: string;
+}
+
+export class HttpDopplerBillingApiClient implements DopplerBillingApiClient {
+  private readonly axios: AxiosInstance;
+  private readonly baseUrl: string;
+  private readonly connectionDataRef: RefObject<AppSession>;
+
+  constructor({
+    axiosStatic,
+    baseUrl,
+    connectionDataRef,
+  }: {
+    axiosStatic: AxiosStatic;
+    baseUrl: string;
+    connectionDataRef: RefObject<AppSession>;
+  }) {
+    this.baseUrl = baseUrl;
+    this.axios = axiosStatic.create({
+      baseURL: this.baseUrl,
+    });
+    this.connectionDataRef = connectionDataRef;
+  }
+
+  private getDopplerBillingApiConnectionData(): DopplerBillingApiConnectionData {
+    const connectionData = this.connectionDataRef.current;
+    if (
+      !connectionData ||
+      connectionData.status !== 'authenticated' ||
+      !connectionData.jwtToken ||
+      !connectionData.userData
+    ) {
+      throw new Error('Doppler Billing API connection data is not available');
+    }
+    return {
+      jwtToken: connectionData.jwtToken,
+      idUser: connectionData.userData.user.idUser,
+    };
+  }
+
+  private mapInvoices(data: any): Invoice[] {
+    return data.map((x: any) => ({
+      accountId: x.accountId,
+      product: x.product,
+      date: x.date,
```

Comment:
If I am not wrong, we are using Dates in place of strings. so, here you should convert the _string date_ to a _Date date_.

 @fcoronelmakingsense could you share some examples about how are we doing that today?

Reply:
Yes, you should return date and not a string (I do not understand why TS is passing):

```ts
export interface Invoice {
  accountId: string;
  product: string;
  date: Date;
  currency: string;
  amount: number;
  filename: string;
}
```

Reply:
Currently the client returns a Date not string for  Date property how you show in the above screenshot

Reply:
We talked with @jhoffman-ms about this and he have applied the related changes. Resolving...